src/unistd.rs

```diff
@@ -2392,3 +2397,425 @@ pub fn access<P: ?Sized + NixPath>(path: &P, amode: AccessFlags) -> Result<()> {
+                #[cfg(not(any(target_os = "android", target_os = "linux")))]
+                expire: (*pw).pw_expire
+            }
+        }
+    }
+}
+
+/// Representation of a Group, based on `libc::group`
+#[derive(Debug, Clone, PartialEq)]
+pub struct Group {
+    /// Group name
+    pub name: String,
+    /// Group ID
+    pub gid: Gid,
+    /// List of Group members
+    pub mem: Vec<String>
+}
+
+impl From<*mut libc::group> for Group {
+    fn from(gr: *mut libc::group) -> Group {
+        unsafe {
+            Group {
+                name: CStr::from_ptr((*gr).gr_name).to_string_lossy().into_owned(),
+                gid: Gid::from_raw((*gr).gr_gid),
+                mem: Group::members((*gr).gr_mem)
+            }
+        }
+    }
+}
+
+impl Group {
+    unsafe fn members(mem: *mut *mut c_char) -> Vec<String> {
+        let mut ret = vec![];
+
+        for i in 0.. {
+            let u = mem.offset(i);
+            if (*u).is_null() {
+                break;
+            } else {
+                let s = CStr::from_ptr(*u).to_string_lossy().into_owned();
+                ret.push(s);
+            }
+        }
+
+        ret
+    }
+}
+
+/// Query a user.
+///
+/// The buffer size variants should not be needed 99% of the time; the default buffer size of 1024
+/// should be more than adequate. Only use the buffer size variants if you receive an ERANGE error
+/// without them, or you know that you will be likely to.
+///
+/// # Examples
+///
+/// ```
+/// use nix::unistd::{User, UserQuery, Queryable};
+/// let find = String::from("root");
+/// // Returns an Option<Result<User>>, thus the double unwrap.
+/// let res = User::query( UserQuery::Name(find.clone()) ).unwrap().unwrap();
+/// assert!(res.name == find);
+/// ```
+#[derive(Debug, Clone, Eq, PartialEq, Hash)]
+pub enum UserQuery {
+    /// Get a user by UID.
+    ///
+    /// Internally, this function calls
+    /// [getpwuid_r(3)](http://pubs.opengroup.org/onlinepubs/9699919799/functions/getpwuid_r.html)
+    Uid(Uid),
+    /// Get a user by name.
+    ///
+    /// Internally, this function calls
+    /// [getpwnam_r(3)](http://pubs.opengroup.org/onlinepubs/9699919799/functions/getpwuid_r.html)
+    Name(String),
+    UidWithBufsize(Uid, usize),
+    NameWithBufsize(String, usize)
+}
+
+/// Query a group.
+///
+/// The buffer size variants should not be needed 99% of the time; the default buffer size of 1024
+/// should be more than adequate. Only use the buffer size variants if you receive an ERANGE error
+/// without them, or you know that you will be likely to.
+///
+/// # Examples
+///
+/// ```ignore
+/// use nix::unistd::{Group, GroupQuery};
+/// // On many systems there's no `root` group; on FreeBSD for example it's known as `wheel`
+/// let find = String::from("root");
+/// // Returns an Option<Result<Group>>, thus the double unwrap. Will panic if there's no `root`
+/// // group.
+/// let res = Group::query( GroupQuery::Name(find.clone()) ).unwrap().unwrap();
```

Comment:
Same comment about the need to `clone` here as above.

Reply:
Done.